source/codegen/metadata_validation.py

```diff
@@ -321,6 +322,15 @@ def _validate_enum(enum_name: str, used_enums: Set[str], metadata: dict):
                     metadata, RULES.ENUMS_SHOULD_NOT_HAVE_DUPLICATE_VALUES, ["enums", enum_name]
                 ):
                     raise Exception(f"Duplicate values in enum!")
+        elif common_helpers.get_driver_readiness(metadata["config"]) == "Release":
+            # TODO AB#1991199: Raise exception instead of warning.
+            warnings.warn(
+                f"{metadata['config']['namespace_component']}::{enum_name} -> Enum is in metadata but is not referenced by a function/attribute or force-included."
+            )
+        else:
+            warnings.warn(
```

Comment:
I'm not gonna rock the boat. I just know that personally, when initially implementing the rfmx drivers, I would've preferred it not be required to have this check pass in my first commit.

Reply:
Yeah, that's fair. After thinking about it more I think I'd want to err on the side of failing regardless to avoid the developer being blindsided later when they want to toggle it on. It's in their hands at the point of failure then to either address it upfront or workaround (force-include or comment out enum) and track fixing it later. I went ahead and switched it to not look at the release readiness.